src/worker/push-worker.js

```diff
@@ -0,0 +1,33 @@
+/*
+  Copyright 2015 Google Inc. All Rights Reserved.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+/* eslint-env browser */
+
+/**
+ * PushWorker is a front end library that simplifies adding push to your
+ * site.
+ */
+export default class PushWorker {
+  getNotifications(tagName) {
+    if (typeof tagName !== 'undefined' && typeof tagName !== 'string') {
+      throw new Error('The optional tagName argument must be a string if set');
+    }
+
+    if (tagName && tagName.length === 0) {
```

Comment:
Got it. Personally, I'll forget about that detail and would prefer the explicit  condition here, otherwise I'll wrote a comment explaining the above.....thoughts?